### PR TITLE
fix(skills): normalize load_skill_from_dir relative path resolution

### LINKED ISSUE #4684: Path Inconsistency for load_skill_from_dir between adk web and Agent Engine Deployment
### Description

There is a discrepancy in how ADK Skills loaded via `load_skill_from_dir` behave in local development versus deployed environments. Specifically, the relative pathing required to load a skill directory works correctly when running locally with `adk web` but fails when the agent is deployed to Agent Engine (AE) because the directory structure changes during the deployment/packaging process.

### Current Behavior

Currently, loading a skill from a directory requires a path relative to the execution root:

```python
my_skill = load_skill_from_dir("agent_name/skills/my-skill-directory")

```

* **Local (`adk web`):** The tool typically runs from the directory above the agent, allowing the relative path to resolve correctly.
* **Deployed (Agent Engine):** When the agent is packaged and deployed to AE, the directory structure is altered or copied, making the hardcoded relative path incorrect and causing the skill to fail to load.

### Expected Behavior

Skills loaded from a directory should resolve consistently across all supported deployment environments. A developer should not need to manually adjust paths or "jump through hoops" to make local code work in production.

### Suggested Improvements

1. **Environment-Agnostic Path Resolution:** Implement a mechanism within the ADK to resolve skill paths relative to the agent's root or package, regardless of the deployment environment.
2. **Automated Packaging/Embedding:** Consider a deployment step that automatically packages or embeds local skill directories (e.g., into Python data classes) to eliminate the dependency on local file paths.
3. **End-to-End Integration Tests:** Expand the test suite to include complex integration tests that verify that documentation examples and directory-based skills work end-to-end in Agent Engine, not just locally.

### PR DESCRIPTION
**Pull request by pandego**
Closes #4684

## Problem
`load_skill_from_dir` currently resolves relative paths from the current working directory only. In Agent Engine flows, code may execute from the agent directory while examples/path inputs include the agent directory prefix (for example `my-agent/skills/my-skill`), producing a duplicated segment and a missing directory error.

## Solution
- Add `_resolve_skill_dir_path` to normalize skill directory resolution for relative paths.
- Support an agent-prefixed relative path when cwd is already the agent directory.
- Reuse this resolver for `load_skill_from_dir`, `validate_skill_dir`, and `read_skill_properties` to keep behavior consistent.
- Add a unit test covering the duplicate-prefix execution case.

## Testing Plan
- `uv run pytest tests/unittests/skills/test__utils.py`

## Checklist
- [x] I have read CONTRIBUTING.md.
- [x] I have performed a self-review.
- [x] I have added tests that prove the fix is effective.
- [x] New and existing unit tests pass locally for touched scope.

### CODE REVIEW

File: src/google/adk/skills/_utils.py [hunk 1]
```diff
@@ -34,6 +34,40 @@
...
+    stripped = pathlib.Path(*path.parts[1:])
+    candidates.append(cwd / stripped)
+
+  candidates.append(cwd.parent / path)
+
+  for candidate in candidates:
+    if candidate.exists():
+      return candidate.resolve()
+
+  return candidates[0].resolve()
```
[the remark this answers is not in the document]
Comment (pandego) [reply src/google/adk/skills/_utils.py]: Good catch, agreed.

I updated `_resolve_skill_dir_path` in `15d498ea` to:
- prioritize the stripped-prefix candidate (`cwd / stripped`) before `cwd / path`
- remove the `cwd.parent / path` fallback
- keep fallback resolution anchored to `cwd / path`

Validation rerun:
- `uv run pytest tests/unittests/skills/test__utils.py` (10 passed)

**Comment by rohityan** [comment]
Hi @pandego , Thank you for your contribution! We appreciate you taking the time to submit this pull request. Can you fix the formatting errors before we can proceed with the review. You can use autoformat.sh.

**Comment by rohityan** [comment]
Hi @DeanChensj , can you please review this. LGTM

**Comment by DeanChensj** [comment]
Hi @wukath , can you help take a look at this PR?

**Comment by wukath** [comment]
Hi @pandego thanks for your PR! We already have a PR for the fix internally that is in review, will let you know when it gets in

**Comment by pandego** [comment]
Thanks for the update - understood.

I will pause on this PR while your internal fix is in review.
If your internal PR lands and fully supersedes this one, I can close this PR to avoid duplicate noise.

**Comment by pandego** [comment]
Thanks for the update here. Since you already have an internal fix in review for the same issue, I am closing this PR to avoid duplicate noise on your side.

If that internal change does not land or you want me to reopen and continue from this branch, I am happy to pick it back up.
